tests/test_utilities.py

```diff
@@ -344,5 +345,18 @@ def test_dont_drop_packages(self) -> None:
         self.assertEqual(frozenset(specs.keys()),
                          frozenset(topological_sort(specs)))
 
+    def test_cycle(self) -> None:
```

Comment:
Thank you for your contribution.

Can you add a few more cases? In particular:

- An empty set of packages
- Three sets of packages which are not dependent on each other
- A proper dependency chain A->B->C
- A diamond topology A->B1 A->B2 B1->C B2->C
- A single package

Reply:
Most of these cases were already covered by older tests, I added tests for an empty set of packages, and three sets of packages which are not dependent on each other